controllers/storagecluster/cephcluster.go

```diff
@@ -727,12 +727,8 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster, serverVersion *version.
 			}
 
 			// Annotation crushDeviceClass ensures osd with different CRUSH device class than the one detected by Ceph
-			crushDeviceClass := ds.DeviceType
```

Comment:
There are still scenarios where other deviceClasses may be set. For now, let's leave the old behavior when replica 1 is disabled, and the new "replicated" device class when it is enabled. We will have to confirm with PM if this is sufficient for the other deviceClass scenarios.

Reply:
done